### PR TITLE
[Release 2.14.0] Add release notes

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Add 2.14.0 release notes
 
### Issues Resolved
Part of https://github.com/opensearch-project/job-scheduler/issues/606
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
@cwperks can you please review this PR?
Adding @bbarani @rishabh6788

**Comment by prudhvigodithi** [comment]
The `codecov/codecov-action` was changed to `v3` as the `opensearch-project/opensearch-build/.github/workflows/get-ci-image-tag.yml@main` was changed back to `CentOS7` (Related [PR](https://github.com/opensearch-project/opensearch-build/pull/4667/files)), the v4 works with no problem with `almalinux8`.
Adding @peterzhuamazon

**Comment by peterzhuamazon** [comment]
Yes @prudhvigodithi expected behavior as CentOS7 does not support node16 or up.

File: release-notes/opensearch-job-scheduler.release-notes-2.14.0.0.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## Version 2.14.0.0
+
+Compatible with OpenSearch 2.14.0
+
+### Maintenance
+* Increment version to 2.14.0 ([#605](https://github.com/opensearch-project/job-scheduler/pull/605))
```
Comment (cwperks) [inline release-notes/opensearch-job-scheduler.release-notes-2.14.0.0.md:6]: nit: Version increment PRs can be left out of release notes.

**Review by cwperks — APPROVED** [review]

**Pull request merged**